### ISSUE TITLE
[APP] Browserstack app-automate appium

### ISSUE THREAD

**Issue by damienleger** [PipedreamHQ/pipedream#2091]
**Name of app / service**
Browserstack app-automate appium

**Link to developer documentation**
https://www.browserstack.com/docs/app-automate/api-reference/appium/apps

**Is lack of support preventing you from building workflows, or do you have a workaround?**
I think no workaround with Browserstack. I would like to monitor new app uploaded by customId and run a Jenkins job with "app_url" returned parameter

**Are there specific actions, or triggers, you'd like to see for this app? Please let us know here or use the Action and Trigger issue templates to open requests for each!**
Trigger on new upload for a given customId. Action to send a custom http query with fields from browserstack answers + custom fields

**Comment by dannyroosevelt** [comment]
@damienleger here's a basic integration for BrowserStack: https://pipedream.com/apps/browserstack.

> Trigger on new upload for a given customId

To confirm, is this the endpoint you'd like to call, at some regular interval? I don't see webhooks supported in their documentation, so this appears to be the next best option: https://www.browserstack.com/docs/app-automate/api-reference/appium/apps#list-uploaded-apps

> Action to send a custom http query with fields from browserstack answers + custom fields

Can you elaborate on this action, or link to their documentation, to make sure I fully understand what you're trying to accomplish?

**Comment by damienleger** [comment]
hello @dannyroosevelt , thank you I was looking for a browserstack trigger so I was looking in the wrong place 🤦‍♂️

I managed to do the browserstack https://api-cloud.browserstack.com/app-automate/recent_apps API call with custom_id get query parameter. I need to get app_url response attribute and use it as parameter to run a job with parameter via Jenkins API (https://www.jenkins.io/doc/book/using/remote-access-api/).

My mobile app is uploaded via CircleCI pipeline to browserstack. I though this was a good idea to decoupling the Jenkins job run from this CircleCI pipeline by using pipedream in between but now I the more I think of it the more I think this is just overcomplicate things for no real benefits 🤔

Closing this ticket then. Sorry for the inconvenience! 🙏
